instaclustr/api_client.go

```diff
@@ -222,3 +222,77 @@ func (c *APIClient) ReadVpcPeering(cdcID string, vpcPeeringID string) (*VPCPeeri
 	json.Unmarshal(bodyText, &vpcPeering)
 	return &vpcPeering, nil
 }
+
+func (c *APIClient) EncryptionKeyAdd(data []byte) (string, error) {
+	url := fmt.Sprintf("%s/provisioning/v1/encryption-keys", c.apiServerHostname)
+	resp, err := c.MakeRequest(url, "POST", data)
+	if err != nil {
+		return "", err
+	}
+
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 202 {
+		return "", errors.New(fmt.Sprintf("Status code: %d, message: %s", resp.StatusCode, bodyText))
+	}
+
+	var respJson interface{}
+	var id string
+	err = json.Unmarshal(bodyText, &respJson)
+	if err != nil {
+		return "", err
+	}
+
+	respJsonData := respJson.(map[string]interface{})
+	for _, value := range respJsonData {
+		id = value.(string)
+	}
+	return id, nil
+}
+
+func getResourceByID(resources *[]EncryptionKey, id string) (*EncryptionKey, error) {
+	for _, resource := range *resources {
+		if resource.ID == id {
+			return &resource, nil
+		}
+	}
+	return nil, errors.New(id)
+}
+
+func (c *APIClient) EncryptionKeyRead(id string) (*EncryptionKey, error) {
+	url := fmt.Sprintf("%s/provisioning/v1/encryption-keys", c.apiServerHostname)
+	resp, err := c.MakeRequest(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	bodyText, err := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != 200 {
+		return nil, errors.New(fmt.Sprintf("Status code: %d, message: %s", resp.StatusCode, bodyText))
+	}
+	var kmsKeys []EncryptionKey
+	err = json.Unmarshal(bodyText, &kmsKeys)
+
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Could not unmarshal JSON - Status code: %d, message: %s", resp.StatusCode, bodyText))
+	}
+
+	keyResource, err := getResourceByID(&kmsKeys, id)
+	if err != nil {
+		return nil, errors.New(fmt.Sprintf("Error encryption key %s does not exist", id))
+	}
+
+	return keyResource, nil
+}
+
+func (c *APIClient) EncryptionKeyDelete(keyID string) error {
```

Comment:
Could this be renamed to DeleteEncryptionKey for consistency with other read methods? Example - DeleteCluster